include/zephyr/kernel.h

```diff
@@ -3606,6 +3606,19 @@ int k_work_queue_drain(struct k_work_q *queue, bool plug);
  */
 int k_work_queue_unplug(struct k_work_q *queue);
 
+/** @brief Stop a work queue.
+ *
+ * This stops the work queue thread and releases its resources. This should be
+ * invoked when the work queue is no longer needed.
```

Comment:
"Releases its resources" is a little unspecific.  This stops a thread, but doesn't actually "release" anything as the thread object remains owned by the queue.

It also wouldn't hurt to explicitly point out that the call is synchronous: the thread is stopped before the call returns so the app is guaranteed that no further handlers will be in progress or called in the future.